src/utils/types.ts

```diff
@@ -24,8 +26,35 @@ export interface UserSignupRequestFail {
     fullMessages: string[];
   };
 }
-
 export interface UserLoginRequestFail {
   status: string;
   errors: string[];
 }
+
+export interface Books {
+  id: number;
+  author: string;
+  title: string;
+  imageUrl: string;
+  publisher: string;
+  year: number;
+  genre: string;
+  createdAt: string;
+  updatedAt: string;
+}
+
+export interface BooksRequestFail {
```

Comment:
this is a generalized error, every request that fails comes with this error interface